src/indexable.cr

```diff
@@ -23,34 +23,33 @@ module Indexable(T)
   # This method should only be directly invoked if you are absolutely
   # sure the index is in bounds, to avoid a bounds check for a small boost
   # of performance.
-  abstract def unsafe_at(index : Int)
+  abstract def unsafe_fetch(index : Int)
 
   # Returns the element at the given *index*, if in bounds,
-  # otherwise executes the given block and returns its value.
+  # otherwise executes the given block with the index and returns its value.
   #
   # ```
   # a = [:foo, :bar]
-  # a.at(0) { :baz } # => :foo
-  # a.at(2) { :baz } # => :baz
+  # a.fetch(0) { :default_value }    # => :foo
+  # a.fetch(2) { :default_value }    # => :default_value
+  # a.fetch(2) { |index| index * 3 } # => 6
   # ```
-  def at(index : Int)
+  def fetch(index : Int)
     index = check_index_out_of_bounds(index) do
-      return yield
+      return yield index
     end
-    unsafe_at(index)
+    unsafe_fetch(index)
   end
 
-  # Returns the element at the given *index*, if in bounds,
-  # otherwise raises `IndexError`.
+  # Returns the value at the index given by *index*, or when not found the value given by *default*.
   #
   # ```
   # a = [:foo, :bar]
-  # a.at(0) # => :foo
-  # a.at(2) # raises IndexError
+  # a.fetch(0, :default_value) # => :default_value
```

Comment:
Shouldn't the return value here be `:foo`?

Reply:
nice find, I guess the example should be `a.fetch(2, :default_value)` instead

Reply:
@bew it's covered one line below.

Reply:
oh ok, then `:foo` yeah

Reply:
Oh yes, sorry, my bad. Fixed.

Reply:
Seems like it's still not fixed.

Reply:
Yep, you're right. Seems something went wrong. Sorry, fixed (I hope!)